### PR TITLE
Fix saving paths to dashboard with horizontal UI

### LINKED ISSUE #4828: User Paths won't save filter to dashboard
## Bug description

Can't save user paths with filteres onto dashboard.

This happens only when feature flag `4050-query-ui-optB` is enabled!

## Expected behavior

Paths on dashboards should keep the right filters.

## How to reproduce

1. Open insights / paths.
2. Add a filter
3. Save to a dashboard
4. Open the dashboard
5. Open the insight
6. See that the filter is missing

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: _please provide_

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Depends on https://github.com/PostHog/posthog/pull/4843
- Closes https://github.com/PostHog/posthog/issues/4828

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] Frontend/CSS is usable at 320px (iPhone SE) and decent at 360px (most phones)
- [ ] Breaking changes are backwards-compatible. Ensure old/new frontend requests work with new/old backends, and vice versa.

### CODE REVIEW

**Comment by mariusandra** [comment]
Merged with master and ready for a look.

Similar issues exist with lifecycle and other graphs (see https://github.com/PostHog/posthog/issues/4787), but I'll chunk this down since I need to AFK for an hour now.

File: frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts [hunk 1]
```diff
@@ -132,6 +132,12 @@ export const propertyFilterLogic = kea<propertyFilterLogicType>({
 
     selectors: {
         filtersLoading: [() => [propertyDefinitionsModel.selectors.loaded], (loaded) => !loaded],
+        filtersNotEmpty: [
+            (s) => [s.filters],
+            (filters) => {
+                return filters.filter((f) => 'key' in f) as PropertyFilter[]
```
Comment (paolodamico) [inline frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts:138]: Not sure if it's worth checking too if `key` is not an empty string.

**Review by paolodamico — APPROVED** [review]
Works as expected, lgtm!

**Review by samwinslow — APPROVED** [review]

**Comment by mariusandra** [comment]
Did some refactoring and extracted the empty/filled checking logic to a function. Will merge if tests pass.

**Pull request merged**